core/blockchain.go

```diff
@@ -1524,15 +1571,15 @@ func (bc *BlockChain) reorg(oldBlock, newBlock *types.Block) error {
 	} else {
 		log.Error("Impossible reorg, please file an issue", "oldnum", oldBlock.Number(), "oldhash", oldBlock.Hash(), "newnum", newBlock.Number(), "newhash", newBlock.Hash())
 	}
-	// Insert the new chain, taking care of the proper incremental order
-	for i := len(newChain) - 1; i >= 0; i-- {
+	// Insert the new chain(except the head block(reverse order)),
+	// taking care of the proper incremental order.
+	for i := len(newChain) - 1; i >= 1; i-- {
```

Comment:
Why aren't we writing `newChain[0]`?

Reply:
Ah, if I understand correctly, we're calling reorg from `writeBlockWithState`, so the injection is done outside already.

Reply:
Exactly.